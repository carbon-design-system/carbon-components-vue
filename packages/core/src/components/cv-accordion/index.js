import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
