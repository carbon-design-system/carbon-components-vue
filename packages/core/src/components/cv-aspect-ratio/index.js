import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
