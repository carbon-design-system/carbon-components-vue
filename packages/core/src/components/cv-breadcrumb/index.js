import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
