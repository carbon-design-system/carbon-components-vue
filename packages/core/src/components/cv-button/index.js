import CvButton from './cv-button';
import CvIconButton from './cv-icon-button';
import CvButtonSkeleton from './cv-button-skeleton';
import CvButtonSet from './cv-button-set';

export { CvButton, CvIconButton, CvButtonSkeleton, CvButtonSet };
