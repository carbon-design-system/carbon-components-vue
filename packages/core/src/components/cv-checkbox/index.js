import CvCheckbox from './cv-checkbox';
import CvCheckboxSkeleton from './cv-checkbox-skeleton';
export { CvCheckbox, CvCheckboxSkeleton };
