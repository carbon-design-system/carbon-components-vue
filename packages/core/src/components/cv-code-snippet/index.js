import CvCodeSnippet from './cv-code-snippet';
import CvCodeSnippetSkeleton from './cv-code-snippet-skeleton';
export { CvCodeSnippet, CvCodeSnippetSkeleton };
