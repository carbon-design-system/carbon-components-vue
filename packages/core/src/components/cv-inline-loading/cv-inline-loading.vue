<template>
  <div data-inline-loading :class="`${carbonPrefix}--inline-loading`" role="alert" aria-live="assertive">
    <div
      :class="[
        `${carbonPrefix}--inline-loading__animation`,
        { [`${carbonPrefix}--loading--stop`]: internalState === STATES.ENDING },
      ]"
    >
      <div
        v-show="internalState === STATES.LOADING || internalState === STATES.ENDING"
        :class="`${carbonPrefix}--loading ${carbonPrefix}--loading--small`"
      >
        <svg :class="`${carbonPrefix}--loading__svg`" viewBox="-75 -75 150 150">
          <circle :class="`${carbonPrefix}--loading__background`" cx="0" cy="0" r="26.8125" />
          <circle :class="`${carbonPrefix}--loading__stroke`" cx="0" cy="0" r="26.8125" />
        </svg>
      </div>
      <CheckmarkFilled16
        :hidden="internalState !== STATES.LOADED"
        :class="`${carbonPrefix}--inline-loading__checkmark-container`"
      />
      <ErrorFilled16 :hidden="internalState !== STATES.ERROR" :class="`${carbonPrefix}--inline-loading--error`" />
    </div>
    <p :class="`${carbonPrefix}--inline-loading__text`">{{ stateText }}</p>
  </div>
</template>

<script>
import { STATES } from './consts';
import ErrorFilled16 from '@carbon/icons-vue/lib/error--filled/16';
import CheckmarkFilled16 from '@carbon/icons-vue/lib/checkmark--filled/16';
import { carbonPrefixMixin } from '../../mixins';

export default {
  name: 'CvInlineLoading',
  components: { ErrorFilled16, CheckmarkFilled16 },
  mixins: [carbonPrefixMixin],
  created() {
    this.STATES = STATES;
  },
  props: {
    active: {
      type: Boolean,
      default: undefined,
      validator: val => {
        if (val !== undefined && process.env.NODE_ENV === 'development') {
          console.warn('CvInlineLoading: active prop deprecated in favour of state prop');
        }
        return true;
      },
    },
    endingText: { type: String, default: 'Load ending...' },
    errorText: { type: String, default: 'Loading data failed.' },
    loadingText: { type: String, default: 'Loading data...' },
    loadedText: { type: String, default: 'Data loaded.' },
    state: {
      type: String,
      default: undefined,
      validator: val => {
        if (Object.keys(STATES).some(state => STATES[state] === val.toLowerCase())) {
          return true;
        } else {
          console.error(`CvInlineLoading: Valid states are ${JSON.stringify(Object.values(STATES))}`);
          return false;
        }
      },
    },
  },
  computed: {
    internalState() {
      if (this.state !== undefined) {
        return this.state.toLowerCase();
      } else {
        return this.active ? STATES.LOADING : STATES.LOADED;
      }
    },
    stateText() {
      switch (this.internalState) {
        case STATES.LOADED:
          return this.loadedText;
        case STATES.ERROR:
          return this.errorText;
        case STATES.ENDING:
          return this.endingText;
        default:
          return this.loadingText;
      }
    },
  },
};
</script>
