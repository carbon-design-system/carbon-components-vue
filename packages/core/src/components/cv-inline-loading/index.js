import CvInlineLoading from './cv-inline-loading';
import { STATES } from './consts';

export { CvInlineLoading, STATES };
