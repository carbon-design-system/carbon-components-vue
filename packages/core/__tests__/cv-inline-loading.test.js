import { testComponent, awaitNextTick } from './_helpers';
const { shallowMount: shallow } = awaitNextTick;
import { CvInlineLoading, STATES } from '@/components/cv-inline-loading';

describe('CvInlineLoading', () => {
  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(CvInlineLoading, ['active'], Boolean);
  testComponent.propsAreType(
    CvInlineLoading,
    ['endingText', 'errorText', 'loadingText', 'loadedText', 'state'],
    String
  );
  testComponent.propsHaveDefault(CvInlineLoading, ['endingText', 'errorText', 'loadingText', 'loadedText']);
  testComponent.propsHaveDefaultOfUndefined(CvInlineLoading, ['active', 'state']);

  it('`state` prop validator works as expected', async () => {
    const wrapper = await shallow(CvInlineLoading);
    expect(
      wrapper.vm.$options.props.state.validator && wrapper.vm.$options.props.state.validator(STATES.LOADED)
    ).toBeTruthy();
    expect(
      wrapper.vm.$options.props.state.validator && wrapper.vm.$options.props.state.validator(STATES.ERROR)
    ).toBeTruthy();
    expect(
      wrapper.vm.$options.props.state.validator && wrapper.vm.$options.props.state.validator(STATES.LOADING)
    ).toBeTruthy();
    expect(
      wrapper.vm.$options.props.state.validator && wrapper.vm.$options.props.state.validator(STATES.ENDING)
    ).toBeTruthy();

    // accept lower case
    expect(
      wrapper.vm.$options.props.state.validator &&
        wrapper.vm.$options.props.state.validator(STATES.LOADED.toUpperCase())
    ).toBeTruthy();
    expect(
      wrapper.vm.$options.props.state.validator && wrapper.vm.$options.props.state.validator(STATES.ERROR.toUpperCase())
    ).toBeTruthy();
    expect(
      wrapper.vm.$options.props.state.validator &&
        wrapper.vm.$options.props.state.validator(STATES.LOADING.toUpperCase())
    ).toBeTruthy();
    expect(
      wrapper.vm.$options.props.state.validator &&
        wrapper.vm.$options.props.state.validator(STATES.LOADED.toUpperCase())
    ).toBeTruthy();

    // suppress the error message from the state validator
    const consoleError = console.error;
    console.error = jest.fn();

    expect(wrapper.vm.$options.props.state.validator && wrapper.vm.$options.props.state.validator('TEST')).toBeFalsy();

    // restore
    console.error = consoleError;
  });

  // ***************
  // SNAPSHOT TESTS
  // ***************

  it('should render correctly when state is LOADED`', async () => {
    const propsData = { state: STATES.LOADED, loadedText: 'loaded text test' };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when state is LOADING`', async () => {
    const propsData = { state: STATES.LOADING, loadingText: 'loading text test' };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when state is ERROR`', async () => {
    const propsData = { state: STATES.ERROR, errorText: 'error text test' };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when state is ENDING`', async () => {
    const propsData = { state: STATES.ENDING, errorText: 'ending text test' };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
  it('`state` prop overrides `active` prop', async () => {
    const propsData = { state: STATES.ERROR, active: true };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.vm.internalState).toEqual(STATES.ERROR);
  });

  it('`active` truthy prop results in LOADING `internalState`', async () => {
    const propsData = { active: true };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.vm.internalState).toEqual(STATES.LOADING);
  });

  it('`active` falsy prop results in LOADED `internalState`', async () => {
    const propsData = { active: false };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.vm.internalState).toEqual(STATES.LOADED);
  });

  it('`stateText` should be equal to `loadedText` prop when state is LOADED', async () => {
    const loadedText = 'loaded text test';
    const propsData = { state: STATES.LOADED, loadedText };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.vm.stateText).toEqual(loadedText);
  });

  it('`stateText` should be equal to `errorText` prop when state is ERROR', async () => {
    const errorText = 'error text test';
    const propsData = { state: STATES.ERROR, errorText };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.vm.stateText).toEqual(errorText);
  });

  it('`stateText` should be equal to `ending` prop when state is ENDING', async () => {
    const endingText = 'ending text test';
    const propsData = { state: STATES.ENDING, endingText };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.vm.stateText).toEqual(endingText);
  });

  it('`stateText` should be equal to `ending` prop when state is LOADING', async () => {
    const loadingText = 'loading text test';
    const propsData = { state: STATES.LOADING, loadingText };
    const wrapper = await shallow(CvInlineLoading, { propsData });
    expect(wrapper.vm.stateText).toEqual(loadingText);
  });
});
