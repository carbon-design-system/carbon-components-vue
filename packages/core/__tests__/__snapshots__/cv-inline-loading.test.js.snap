// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvInlineLoading should render correctly when state is ENDING\` 1`] = `
<div data-inline-loading="" role="alert" aria-live="assertive" class="bx--inline-loading">
  <div class="bx--inline-loading__animation bx--loading--stop">
    <div class="bx--loading bx--loading--small">
      <cv-loading-stub active="true" description="Loading" small="true"></cv-loading-stub>
    </div>
    <checkmarkfilled16-stub hidden="true" class="bx--inline-loading__checkmark-container"></checkmarkfilled16-stub>
    <errorfilled16-stub hidden="true" class="bx--inline-loading--error"></errorfilled16-stub>
  </div>
  <p class="bx--inline-loading__text">Load ending...</p>
</div>
`;

exports[`CvInlineLoading should render correctly when state is ERROR\` 1`] = `
<div data-inline-loading="" role="alert" aria-live="assertive" class="bx--inline-loading">
  <div class="bx--inline-loading__animation">
    <div class="bx--loading bx--loading--small" style="display: none;">
      <cv-loading-stub active="true" description="Loading" small="true"></cv-loading-stub>
    </div>
    <checkmarkfilled16-stub hidden="true" class="bx--inline-loading__checkmark-container"></checkmarkfilled16-stub>
    <errorfilled16-stub class="bx--inline-loading--error"></errorfilled16-stub>
  </div>
  <p class="bx--inline-loading__text">error text test</p>
</div>
`;

exports[`CvInlineLoading should render correctly when state is LOADED\` 1`] = `
<div data-inline-loading="" role="alert" aria-live="assertive" class="bx--inline-loading">
  <div class="bx--inline-loading__animation">
    <div class="bx--loading bx--loading--small" style="display: none;">
      <cv-loading-stub active="true" description="Loading" small="true"></cv-loading-stub>
    </div>
    <checkmarkfilled16-stub class="bx--inline-loading__checkmark-container"></checkmarkfilled16-stub>
    <errorfilled16-stub hidden="true" class="bx--inline-loading--error"></errorfilled16-stub>
  </div>
  <p class="bx--inline-loading__text">loaded text test</p>
</div>
`;

exports[`CvInlineLoading should render correctly when state is LOADING\` 1`] = `
<div data-inline-loading="" role="alert" aria-live="assertive" class="bx--inline-loading">
  <div class="bx--inline-loading__animation">
    <div class="bx--loading bx--loading--small">
      <cv-loading-stub active="true" description="Loading" small="true"></cv-loading-stub>
    </div>
    <checkmarkfilled16-stub hidden="true" class="bx--inline-loading__checkmark-container"></checkmarkfilled16-stub>
    <errorfilled16-stub hidden="true" class="bx--inline-loading--error"></errorfilled16-stub>
  </div>
  <p class="bx--inline-loading__text">loading text test</p>
</div>
`;
